src/main/core/shd-slave.c

```diff
@@ -27,6 +27,9 @@ struct _Slave {
     /* paths to programs (i.e. shadow plug-ins) that are run by virtual processes */
     GHashTable* programPaths;
 
+    /* start symbols for programs (i.e. shadow plug-ins) */
+    GHashTable* programStartSymbols;
```

Comment:
Ideally, the start symbols and the paths would be stored in the same object and stored in a single hash table keyed by the name, rather than in two separate hash tables.